pkg/spec/spec.go

```diff
@@ -20,7 +20,7 @@ const Specfile = "spec.json"
 
 // ParseDir parses the given environments `spec.json` into a `v1alpha1.Environment`
 // object with the name set to the directories name
-func ParseDir(baseDir, path string) (*v1alpha1.Environment, error) {
+func ParseDir(baseDir, path string, namespace string) (*v1alpha1.Environment, error) {
```

Comment:
You could get around introducing another argument here, by using `jpath.FsDir` (badly named, ik)
https://github.com/grafana/tanka/blob/22de89f3176d3f16e6055dd4329239a1cff791f9/pkg/jsonnet/jpath/dirs.go#L109

That function takes a path and trims the filename from the end, if it exists.
You could then pass the full filename as an argument, and trim the file part away for legacy compat

Reply:
Can you elaborate? It looks like we need to push all the jpath logic into ParseDir. It would end up having `ParseDir(path string)` as a signature. Is that acceptable?